src/shims/panic.rs

```diff
@@ -161,7 +161,24 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
     }
 
-    /// Starta a panic in the interpreter with the given message as payload.
+    /// Handler that should be called when unsupported functionality is encountered.
+    /// This function will either panic within the context of the emulated application
+    /// or return an error in the Miri process context
+    ///
+    /// Return value of `Ok(bool)` indicates whether execution should continue.
+    fn handle_unsupported<S: AsRef<str>>(&mut self, error_msg: S) -> InterpResult<'tcx, ()> {
```

Comment:
Is this the right file for this function?

Reply:
I think `helpers.rs` is probably the better file.

Reply:
I tried to move it, but since we rely on `start_panic` we get a compiler error:

```rust
   Compiling miri v0.1.0 (S:\development\miri)
error[E0599]: no method named `start_panic` found for mutable reference `&mut Self` in the current scope
   --> src\helpers.rs:641:18
    |
641 |             self.start_panic(error_msg.as_ref(), StackPopUnwind::Skip)?;
    |                  ^^^^^^^^^^^ method not found in `&mut Self`
    |
    = help: items from traits can only be used if the type parameter is bounded by the trait
help: the following trait defines an item `start_panic`, perhaps you need to add another supertrait for it:
    |
45  | pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx> + shims::panic::EvalContextExt {
```

I don't think adding this trait bound is appropriate, so I'll leave it in `panic.rs` unless this is ok (or there's a way around this).

Reply:
It should be `this.start_panic`, then it should work in the other file.